overlays/low-resource/README.md

```diff
@@ -0,0 +1,5 @@
+# Low Resources overlay
```

Comment:
From @bobheadxi 
> should we name this low-resource or something more generic?

The generic name feels a bit misleading considering we have buildkite-specific modifications in here. Not need to change it back since this README does explain the usage. Just thinking a customer would see this and think "hey I want to use fewer resources too!"

Reply:
Ah, good catch - I didn't realize there were buildkite-specific volume mounts here. I wonder if they have to be named `buildkite` and `ci`, is this specific to how agents are set up?